src/settings/plugins/ifcfg-rh/nms-ifcfg-rh-reader.c

```diff
@@ -1995,11 +1995,15 @@ make_tc_setting (shvarFile *ifcfg)
 			break;
 
 		qdisc = nm_utils_tc_qdisc_from_str (value, &local);
-		if (!qdisc)
-			PARSE_WARNING ("ignoring bad qdisc: '%s': %s", value, local->message);
+		if (!qdisc) {
+			PARSE_WARNING ("ignoring bad tc qdisc: '%s': %s", value, local->message);
+			continue;
+		}
 
 		if (!nm_setting_tc_config_add_qdisc (s_tc, qdisc))
-			PARSE_WARNING ("duplicate qdisc");
+			PARSE_WARNING ("duplicate tc qdisc");
```

Comment:
another pre-existing bug here is:

Neither `nm-setting-tc-config.c:set_property()` nor `nm-setting-tc-config.c:verify()` has the behavior of rejecting duplicate entries. That means, the API `nm_setting_tc_config_add_qdisc()` does not allow to create such configurations, that are otherwise creatable and valid. Also, ifcfg-writer would write them, ifcfg-reader could not read them back. I think the fix is simply to not let the add function reject duplicates.

Reply:
I don't think a configuration with duplicate qdiscs or filters can be valid. @lkundrak ?

If so, the TC setting should implement verify()	and reject duplicate entries.